src/main.tf

```diff
@@ -0,0 +1,61 @@
+locals {
+  create_accepter = var.accepter_vnet_id != null ? true : false
+
+  requester_vnet_id   = var.requester.data.infrastructure.id
+  requester_vnet_name = element(split("/", local.requester_vnet_id), index(split("/", local.requester_vnet_id), "virtualNetworks") + 1)
+  requester_vnet_rg   = element(split("/", local.requester_vnet_id), index(split("/", local.requester_vnet_id), "resourceGroups") + 1)
+
+  accepter_vnet_id   = try(var.accepter.data.infrastructure.id, var.accepter_vnet_id)
+  accepter_vnet_name = element(split("/", local.accepter_vnet_id), index(split("/", local.accepter_vnet_id), "virtualNetworks") + 1)
+  accepter_vnet_rg   = element(split("/", local.accepter_vnet_id), index(split("/", local.accepter_vnet_id), "resourceGroups") + 1)
+}
+
+data "azurerm_virtual_network" "requester" {
+  name                = local.requester_vnet_name
+  resource_group_name = local.requester_vnet_rg
+}
+
+data "azurerm_virtual_network" "md-accepter" {
+  count               = local.create_accepter ? 0 : 1
+  name                = local.accepter_vnet_name
+  resource_group_name = local.accepter_vnet_rg
+}
+
+data "azurerm_virtual_network" "external-accepter" {
```

Comment:
Do we need 2 different data lookups here? If we're just doing a data lookup we don't care whether we own it or whether its external.